src/containerapp/azext_containerapp/tests/latest/test_containerapp_commands.py

```diff
@@ -561,6 +561,34 @@ def test_containerapp_revision_label_e2e(self, resource_group):
         self.assertEqual(len([w for w in traffic_weight if "label" in w]), 0)
 
 
+class ContainerappAnonymousRegistryTests(ScenarioTest):
+    @AllowLargeResponse(8192)
+    @ResourceGroupPreparer(location="northeurope")
+    @live_only()  # encounters 'CannotOverwriteExistingCassetteException' only when run from recording (passes when run live)
+    def test_containerapp_anonymous_registry(self, resource_group):
```

Comment:
In general can we work with core-cli to figure out getting these tests run without live mode. Given extension tests don't run in nightly or release builds, or PR builds, these don't help until a dev runs all tests locally.
The containerapp-compose is supposed to have running these kind of tests, using a monkey patch of some sort, can we do something similar?

Reply:
The problem we bumped into was the generated name for the log analytics workspace.

We worked around this by adding [a hidden/suppressed parameter](https://github.com/Azure/azure-cli-extensions/blob/main/src/containerapp-compose/azext_containerapp_compose/_params.py#L24) and then using the `self.create_random_name` to pass in a known value (known to the test harness) for the name of the log analytics workspace.

This required a monkey patch of `_generate_log_analytics_workspace_name` by https://github.com/Azure/azure-cli-extensions/blob/main/src/containerapp-compose/azext_containerapp_compose/_monkey_patch.py#L57-L60

I'd be happy to grab some time to pair to see if that'd be helpful for your tests.

Reply:
Thanks @smurawski - given this a create command that can take an existing log-analytics as an input I am curious why this failing. but to your point yes, happy to engage with you as needed.

Reply:
@panchagnula - in this test (see line 575), it's doing a `containerapp env create` which generates a log analytics workspace with a random name.  This will fail on any attempt to replay the recorded test.

Reply:
Thanks for the suggestion @smurawski! I'm testing this right now. Thankfully, we can pass in log analytics workspace names to `az containerapp env create` without needing to add hidden parameters to any of our commands, so this might be an easy fix for us